### PR TITLE
Feature/dex 546 surcharge

### PR DESCRIPTION
**Pull request by EvansPie**
DEX-546

# What this PR does

Implements surcharge UI

# Notable decisions & other stuff

Hardcodes values

# Instructions on how to test this

Let other reviewers know how they can test this.

# Before merging

_QA_

- [x] I manually tested this with a demo application (simulator)
- [ ] I manually tested this with a demo application (real device)
- [ ] I wrote unit tests for each new util-like function
- [ ] I wrote at least basic e2e tests for the new functionalities or new paths
- [ ] If this PR introduces UI changes, I manually tested this PR on iPhone 8 (iOS 10, 12, 14), iPhone 12, and iPhone 12 Max
- [ ] If this PR modifies the behavior of an API, I did my best to make my new API backward compatible
- [ ] If this PR modifies the API, I attempted to implement the feature with an example

_Documentation_

- [ ] If this PR modifies the API, I updated the API Reference.
- [ ] If this PR adds new options to the API, I wrote some documentation or guide in the online documentation
- [ ] If the PR modifies the API, I wrote some documentation to deprecate it

# After merging

- Make sure a new release has been pushed to Cocoapods trunk

### CODE REVIEW

File: Sources/PrimerSDK/Classes/User Interface/Root/PrimerVaultManagerViewController.swift [hunk 1]
```diff
@@ -29,77 +29,47 @@ internal class PrimerVaultManagerViewController: PrimerFormViewController {
...
-                default:
-                    break
-                }
+        let checkoutViewModel: VaultCheckoutViewModelProtocol = DependencyContainer.resolve()
+        var availablePaymentMethods = checkoutViewModel.availablePaymentOptions.filter({ $0.type != .applePay })
+        
+        for (index, paymentMethod) in availablePaymentMethods.enumerated() {
+            switch paymentMethod.type {
+            case .klarna:
+                availablePaymentMethods[index].surCharge = "+£5.36"
```
Comment (xevious78) [inline Sources/PrimerSDK/Classes/User Interface/Root/PrimerVaultManagerViewController.swift:45]: Hardcoded data? How do you plan to get this data from /configuration?

**Review by xevious78 — COMMENTED** [review]
some comments :)

File: Sources/PrimerSDK/Classes/User Interface/Buttons/PaymentMethodButtonView.swift [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+            } else {
+                return nil
+            }
+        }
+        set {
+            button.layer.borderColor = newValue?.cgColor
+        }
+    }
+    
+    override var tintColor: UIColor! {
```
Comment (ceriksson) [inline Sources/PrimerSDK/Classes/User Interface/Buttons/PaymentMethodButtonView.swift:47]: what does this do?

File: Sources/PrimerSDK/Classes/User Interface/Buttons/PaymentMethodButtonView.swift [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    @IBInspectable var borderWidth: CGFloat {
+        get {
+            return button.layer.borderWidth
+        }
+        set {
+            button.layer.borderWidth = newValue
+        }
+    }
+    
+    @IBInspectable var borderColor: UIColor? {
```
Comment (ceriksson) [inline Sources/PrimerSDK/Classes/User Interface/Buttons/PaymentMethodButtonView.swift:34]: we may need to add the option to handle multiple states: error, selected, unselected

File: Sources/PrimerSDK/Classes/User Interface/PaymentMethodsGroupView.swift [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+            paymentMethodButtonView.cornerRadius = 4.0
+            paymentMethodButtonView.borderWidth = 1.0
+            paymentMethodButtonView.borderColor = .clear
+            paymentMethodButtonView.clipsToBounds = true
+            
+            switch paymentMethodViewModel.type {
+            case .paymentCard:
+                paymentMethodButtonView.setTitleColor(theme.colorTheme.text1, for: .normal)
+                paymentMethodButtonView.tintColor = theme.colorTheme.text1
+                paymentMethodButtonView.buttonColor = .white
```
Comment (ceriksson) [inline Sources/PrimerSDK/Classes/User Interface/PaymentMethodsGroupView.swift:75]: we need to discuss taking all these colors and placing them in the theme or similar

**Review by ceriksson — COMMENTED** [review]
looks good, just want to resolve some of the comments

File: Sources/PrimerSDK/Classes/User Interface/Root/PrimerVaultManagerViewController.swift (hunk 1 above)
Comment (EvansPie) [reply Sources/PrimerSDK/Classes/User Interface/Root/PrimerVaultManagerViewController.swift:45]: Yeah, I've mentioned that on the description, I didn't have data back then. This has changed on the other branch that I was working with the local backend.

File: Sources/PrimerSDK/Classes/User Interface/Root/PrimerVaultManagerViewController.swift [hunk 5]
```diff
@@ -29,77 +29,47 @@ internal class PrimerVaultManagerViewController: PrimerFormViewController {
...
+            case .paymentCard:
+                availablePaymentMethods[index].surCharge = "Additional fee may apply"
```
[the remark this answers is not in the document]
Comment (EvansPie) [reply Sources/PrimerSDK/Classes/User Interface/Root/PrimerVaultManagerViewController.swift:47]: Done

File: Sources/PrimerSDK/Classes/User Interface/Buttons/PaymentMethodButtonView.swift (hunk 3 above)
Comment (EvansPie) [reply Sources/PrimerSDK/Classes/User Interface/Buttons/PaymentMethodButtonView.swift:34]: Agreed, but this isn't the scope of this PR, I believe it's within the theming scope.

**Pull request merged**